### ISSUE TITLE
_forget() support for AMQPBackend

### ISSUE THREAD

**Issue by Hexadite-Max** [celery/celery#2704]
Hi,

Couldn't find anywhere a reference for why "_forget" is not implemented in the AMQPBackend, proposed solution:

```
class AMQPBackend(BaseBackend):
    ...
    def _forget(self, task_id):
        with self.app.pool.acquire_channel(block=True) as (_, channel):
            binding = self._create_binding(task_id)(channel)
            binding.delete(if_unused=False, if_empty=False, nowait=True)
    ...
```

Of course this might require some checks, and override in the RPC, but is there any reason this was not implemented?

**Comment by thedrow** [comment]
I don't really know what this method does.
We'll have to wait for @ask to respond about this one.

**Comment by Hexadite-Max** [comment]
Thnx, for the reply.
This method in other backends deletes the result from the backend store, these are already supported:
* KeyValueStoreBackend (and all subclasses)
* DatabaseBackend
* MongoBackend

\+ i think this should be labeled under https://github.com/celery/celery/labels/Component%3A%20AMQP%20Results%20Backend

**Comment by ask** [comment]
I don't think you can implement this for the RPC backend.  It's possible to do for the old amqp result backend, but that backend should be avoided I think (due to performance limitations).

I guess we can include it for completeness, but what are you using `forget()` for exactly?  It seems like something you don't need that often.

**Comment by Hexadite-Max** [comment]
I also don't think this is appropriate for RPC, however in one of my scenarios I had low frequency tasks (~10 a minute) that needed to post back updates (multiple per task). I didn't want to use a polling backend for this, so I went with the simple AMQPBackend. After each tasks progress is complete (after a few results), the results are no longer needed and can be discarded. However without calling __forget()__ these get cleaned up only after the specified queue expiration which by default is 24 hours. This even with 10 new tasks a minute constant rate will eventually build up a backlog of 14400 queues.
So my solution was to inherit the AMQPBackend and implement the ```_forget()``` so that one all update for a task is complete the queue is deleted explicitly - naturally if the process crashes it still relies on the expiration for cleanup.

If you agree, I can PR the required changes.

**Comment by ask** [comment]
Closing this, as we don't have the resources to complete this task.
